specs/ClearBrowsingData.md

```diff
@@ -0,0 +1,268 @@
+                    -> HRESULT {
+                    LPCWSTR result = error == S_OK ? L"Succeeded" : L"Failed";
+                    RunAsync([this, result]() {
+                        MessageBox(nullptr, result, L"Clear Browsing Data", MB_OK);
+                    });
+                    return S_OK;
+                })
+                .Get()));
+    }
+}
+```
+
+### .NET, WinRT
+```c#
+
+private void ClearAutofillData() 
+{
+    CoreWebView2Profile profile;
+    if (webView.CoreWebView2 != null) 
+    {
+        profile = webView.CoreWebView2.Profile;
+        double endTime = DateTimeOffset.Now.ToUnixTimeSeconds();
+        // Get the current time which is the time in which the browsing data will be cleared until.
+        double startTime = endTime - 3600;
+        // Offset the current time by 3600 seconds to clear the browsing data from the last hour.
+        CoreWebView2BrowsingDataKinds dataKinds = (CoreWebView2BrowsingDataKinds)(CoreWebView2BrowsingDataKinds.GeneralAutofill | CoreWebView2BrowsingDataKinds.PasswordAutosave);
+        await profile.ClearBrowsingDataAsync(dataKinds, startTime, endTime);
+    }
+}
+
+```
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2ClearBrowsingDataCompletedHandler;
+
+/// Specifies the datatype for the
+/// `ICoreWebView2Profile::ClearBrowsingData` method.
+[v1_enum]
+typedef enum COREWEBVIEW2_BROWSING_DATA_KINDS {
+  /// Specifies data stored by the AppCache DOM feature.
+  COREWEBVIEW2_BROWSING_DATA_KINDS_APP_CACHE = 1 << 0,
+
+  /// Specifies file systems data.
+  COREWEBVIEW2_BROWSING_DATA_KINDS_FILE_SYSTEMS = 1 << 1,
+
+  /// Specifies data stored by the IndexedDB DOM feature.
+  COREWEBVIEW2_BROWSING_DATA_KINDS_INDEXED_DB = 1 << 2,
+
+  /// Specifies data stored by the localStorage DOM API.
+  COREWEBVIEW2_BROWSING_DATA_KINDS_LOCAL_STORAGE = 1 << 3,
+
+  /// Specifies data stored by the Web SQL database DOM API.
+  COREWEBVIEW2_BROWSING_DATA_KINDS_WEB_SQL = 1 << 4,
+
+  /// Specifies cache storage which stores the network requests
+  /// and responses. 
+  COREWEBVIEW2_BROWSING_DATA_KINDS_CACHE_STORAGE = 1 << 5,
+
+  /// Specifies DOM storage data. This browsing data kind is inclusive 
+  /// of COREWEBVIEW2_BROWSING_DATA_KINDS_APP_CACHE, 
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_FILE_SYSTEMS,
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_INDEXED_DB, 
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_LOCAL_STORAGE,
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_WEB_SQL, 
+  /// COREWEBVIEW2_BROWSING_DATA_KINDS_CACHE_STORAGE.
+  COREWEBVIEW2_BROWSING_DATA_KINDS_DOM_STORAGE = 
+      COREWEBVIEW2_BROWSING_DATA_KINDS_APP_CACHE|
+      COREWEBVIEW2_BROWSING_DATA_KINDS_FILE_SYSTEMS | 
+      COREWEBVIEW2_BROWSING_DATA_KINDS_INDEXED_DB |
+      COREWEBVIEW2_BROWSING_DATA_KINDS_LOCAL_STORAGE | 
+      COREWEBVIEW2_BROWSING_DATA_KINDS_WEB_SQL |
+      COREWEBVIEW2_BROWSING_DATA_KINDS_CACHE_STORAGE,
```

Comment:
What if a new type of DOM storage is added in the future (e.g., "spatial anchors")? Should `DOM_STORAGE` be a separate flag so you can say "All DOM storage, now and future"?

Reply:
It could be a compat issue either way, if the end developer wanted DOM_STORAGE to only mean the things it means currently and is broken when we start clearing more in the future, or the end developer wanted DOM_STORAGE to mean now and future and is broken when their app is no longer clearing everything.

What is the precedent here? I'm fine with making DOM_STORAGE and SITE, and PROFILE be their own flags that can be updated in the future if we think that's what end developers will expect.

Reply:
Give each collection of bit enum values their own dedicated single bit. And update the docs to make it clear that in the future additional entries may show up. Add 'All' as prefix:

_ALL_DOM_STORAGE
_ALL_PROFILE
_ALL_SITE